vpr/src/device/rr_graph_builder.h

```diff
@@ -58,6 +61,22 @@ class RRGraphBuilder {
 
     /** @brief Clear all the underlying data storage */
     void clear();
+    /** @brief reorder all the nodes
+     * Reorder RRNodeId's using one of these algorithms:
```

Comment:
Here's some high-level context you can add to this comment.

Reordering the rr-graph nodes may be helpful in increasing cache locality during routing, and hence improving compile time. So far, however, the best re-ordering algorithm (DEGREE_BFS) does not speed up the router on most architectures vs. using the node ordering created by the rr-graph builder in VPR, so it is off by default.  The other use of this algorithm is for some unit tests; by changing the order of the nodes in the rr-graph before routing we check that no code depends on the rr-graph node order (as it shouldn't be).

Reply:
Thanks @vaughnbetz. @Raza-jafari-RS Would you mind adding the comments to the API?

Reply:
comment added